### ISSUE TITLE
Potential evaluation leakage in inference pipeline due to use of labels in collate_fn

### ISSUE THREAD

**Issue by Ankeshansh** [abhijitmishra/visual_instruction_rewriting#4]
Hi,

While reproducing the baseline results, I noticed a potential issue in the evaluation pipeline, mainly the test_revision.py.

The current DataLoader uses:
    collate_fn = dataset.collate_fn

Inside this collate function, the processor is called with:
    suffix = labels

This means that during inference, the ground truth responses are being passed into the model input.

As a result:
- The model appears to reproduce the ground truth in the generated output
- This may lead to artificially inflated evaluation metrics

To verify this, I modified the collate function to exclude the suffix during inference, and observed that:
- The duplicated "assistant" outputs disappeared
- Model outputs became more realistic
- Evaluation metrics changed accordingly

Proposed fix:
Use a separate collate function for inference that excludes labels, e.g.:
    def inference_collate_fn(...):
        processor(text=texts, images=images)

I wanted to confirm:
- Is this behavior intentional?
- Or should inference be separated from training in this way?

Additionally, I observed that the generated outputs often contain conversational artifacts such as:
    "system", "user", "assistant"

When evaluation metrics (BLEU, ROUGE, METEOR) are computed directly on these raw outputs, the scores are significantly lower.

After applying simple post-processing to extract only the final assistant response, the evaluation metrics improve noticeably and better reflect the model’s actual performance.

This suggests that evaluation should ideally be performed on cleaned outputs rather than raw generated text.

I wanted to check:
- Is there a recommended post-processing step for evaluation?
- Or should this be incorporated into the evaluation pipeline?

Happy to contribute a PR if this is indeed unintended.

Thanks!

**Comment by abhijitmishra** [comment]
1. To the best of our knowlegde model.generate() ignores labels unless the model is customized to consider labels in eval mode. Teacher forcing is disabledduring generation. But I agree that a cleaner implementation is the one that does not add labels in the DataLoader. 

2. I think we did some offline clearning before evaluating to eliminate specific messages. 

Feel free to raise a PR through suggesting cleanups.
